File: examen/app/src/main/java/com/example/examen1/Marca.kt
package com.example.examen1

class Marca(
    var nombre:String,
    var pais_origen:String,
    var anio_creacion:Int,
    var sucursal_local:Boolean,
    var valor:Float
){
    override fun toString(): String {
        return "Marca(Nombre='$nombre', Pais Origen='$pais_origen', Año Creacion='$anio_creacion', Sucursal Local=$sucursal_local, Valor=$valor.)"
    }
}

File: 03-backend/mascotas/api/models/Marca.js
/**
 * Marca.js
 *
 * @description :: A model definition represents a database table/collection.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {
   nombre: {
      type:"string",
      unique: true,
      required: true
    },
    pais_origen:{
      type:"string",
      required: true
    },
    anio_creacion: {
      type:"number",
      required: true
    },
    sucursal_local: {
      type:"boolean",
      required: true
    },
    valor: {
      type:"number",
      required: true
    },
    modelo: { // One to Many (plural)
      collection: 'modelo', // Referencia al modelo
      via: 'marca' // Nombre Foreign Key en 'Pokemon'
    }
  },
};
